tck/src/main/java/org/eclipse/microprofile/context/tck/ThreadContextTest.java

```diff
@@ -1426,4 +1412,122 @@ public void currentContextExecutorRunsWithContext() throws InterruptedException,
             Label.set(null);
         }
     }
+
+    /**
+     * Verify that the presence of a default executor service allows contextualised CompletableFuture/CompletionStage
+     * to run async actions on them, and that they are running on that executor service.
+     *
+     * @throws ExecutionException indicates test failure
+     * @throws InterruptedException indicates test failure
+     * @throws TimeoutException indicates test failure
+     */
+    @Test
+    public void withDefaultExecutorServiceContextCanInvokeAsyncActions()
+            throws ExecutionException, InterruptedException, TimeoutException {
+        ExecutorService executorService = Executors.newSingleThreadExecutor(new TckThreadFactory());
+        ContextManager contextManager = ContextManagerProvider
+                .instance()
+                .getContextManagerBuilder()
```

Comment:
Same as previous comment - getContextManagerBuilder is optional, so test needs to skip if not implemented.